craft_store/base_client.py

```diff
@@ -186,10 +186,23 @@ def upload_file(
         """Upload filepath to storage.
 
         The monitor_callback is a method receiving one argument of type
-        ``MultipartEncoderMonitor``, among others, it's ``.bytes_read`` and
-        ``.len`` attributes can be used to track progress.
+        ``MultipartEncoder``, this method is to return a method to track
+        progress. Among others, it's ``.bytes_read`` and ``.len`` attributes
+        can be used to track progress. The simplest implementation can look like:
+
+        .. code-block:: python
+
+          def monitor_callback(encoder: requests_toolbelt.MultipartEncoder):
+
+              # instantiate progress class with total bytes encoder.len
+
+              def progress_printer(monitor: requests_toolbelt.MultipartEncoderMonitor):
+                 # Print progress using monitor.bytes_read
```

Comment:
Ah, it's the argument to the method returned by `monitor_callback`. So maybe we could say something like: "...return a method that receives an argument of type ``MultipartEncoderMonitor``. This argument contains the ``.bytes_read`` and ``.len`` attributes that are used to track progress"?

Reply:
updated with class information for clarity